### DISCUSSION TITLE
Restore mask mode in Light2D

### DISCUSSION THREAD

**Discussion by arabgamedev** [godotengine/godot-proposals#7390] [2D]
### Describe the project you are working on

i hope you restore mask mode because it was very handy 
for example you can make among us effect for hiding objects behind walls
like this image
![maxresdefault](https://github.com/godotengine/godot-proposals/assets/66444993/0448f460-fbe2-4667-b38d-d8a87ef72ebf)

### Describe the problem or limitation you are having in your project

can't make this effect without mask mode shadows

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

.

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

.

### If this enhancement will not be used often, can it be worked around with a few lines of script?

.

### Is there a reason why this should be core and not an add-on in the asset library?

.

**Comment by AThousandShips** [comment]
The headings in this are not optional, you have to fill them all in, converting to a discussion, please open discussions instead of incomplete issues in the future
